pbr/scene/goal.py

```diff
@@ -112,16 +112,19 @@ def hide_render(self, to_hide):
     def join_objs(self, objs):
         # Create copy of context
         context = bpy.context.copy()
+
         # Select our active and selected objects
         context["active_object"] = objs[0]
         context["selected_objects"] = objs
+
         # Select all of our available editable bases
         context["selected_editable_bases"] = [
             bpy.context.scene.object_bases[obj.name] for obj in objs
         ]
+
         # Join objects
         bpy.ops.object.join(context)
-        bpy.data.objects[objs[0].name].select = True
```

Comment:
For some reason `bpy.data.objects[objs[0].name].select = True` didn't work for me, as the object with that name wasn't present after the join (does `join()` guarantee that the resulting object will have the name of the first object in the group? If so it doesn't seem to be doing that).

Reply:
Yeah not sure about this one (and why it's never been a problem before), what were the remaining names after the join? 

Might be better as something like this:
```python
# Join objects
names_before = bpy.data.objects.keys()
bpy.ops.object.join(context)
new_name = list(bpy.data.objects.keys() - names_before)[0]
bpy.data.objects[new_name].select = True
```